shape-detection/resources/endtoend-helpers.js

```diff
@@ -0,0 +1,90 @@
+function imageLoadedPromise(image) {
+    return new Promise(function(resolve, reject) {
+        if (image.complete)
+            resolve();
+        image.addEventListener("load", function() {
+            resolve();
+        });
+    });
+}
+
+function videoLoadedPromise(video) {
+    return new Promise(function(resolve, reject) {
+        video.addEventListener("loadeddata", function() {
+            resolve();
+        });
+    });
+}
+
+function waitForNFrames(n) {
+    return new Promise(function(resolve, reject) {
+        var count = n;
+        function tick() {
+            --count;
+            if (count == 0)
+                resolve();
+            else
+                requestAnimationFrame(tick);
+        }
+        requestAnimationFrame(tick);
+    });
+}
+
+function seekTo(video, time) {
+    return new Promise(function(resolve, reject) {
+        async function handleEvent() {
+            video.removeEventListener("seeked", handleEvent);
+            await waitForNFrames(3);
+            resolve();
+        }
+        video.addEventListener("seeked", handleEvent);
```

Comment:
Yeah, wondering the same...

Reply:
As an aside, there is some event watching infra available for tests... I don't know if it will be helpful here or just add indirection, but worth considering.

Reply:
I added a comment :)